mcsema/BC/Function.cpp

```diff
@@ -260,6 +303,123 @@ static llvm::BasicBlock *GetOrCreateBlock(TranslationContext &ctx,
+    }
+
+    auto lifted_func = gModule->getFunction(cfg_func->lifted_name);
+    lifted_func->addFnAttr(llvm::Attribute::UWTable);
+    lifted_func->addFnAttr(llvm::Attribute::OptimizeNone);
+    lifted_func->removeFnAttr(llvm::Attribute::NoUnwind);
+#if LLVM_VERSION_NUMBER > LLVM_VERSION(3, 6)
+    lifted_func->setPersonalityFn(personality);
+#endif
+  }
+
+  for (auto &entry : cfg_func->eh_frame) {
+    if (!entry->lp_ea) {
+      continue;
+    }
+
+    if (!entry->action_index) {
+      std::stringstream ss;
+      ss << "cleanup_landingpad_" << std::hex << entry->lp_ea;
+      auto landing_bb = llvm::BasicBlock::Create(*gContext, ss.str(), ctx.lifted_func);
+      llvm::IRBuilder<> ir(landing_bb);
+      auto exn_type = llvm::StructType::get(llvm::Type::getInt8PtrTy(*gContext),
+                                            llvm::Type::getInt32Ty(*gContext), nullptr);
+#if LLVM_VERSION_NUMBER > LLVM_VERSION(3, 6)
+      auto lpad = ir.CreateLandingPad(exn_type, 1, "cleanup.lpad");
+#else
+      auto personality = gModule->getFunction(PERSONALITY_FUNCTION);
+      auto lpad = ir.CreateLandingPad(exn_type, personality, 1, "cleanup.lpad");
+#endif
+      lpad->setCleanup(true);
+
+      auto exct_value_0  = ir.CreateExtractValue(lpad, 0);
+      auto store_loc_0 = ir.CreateAlloca(exct_value_0->getType());
+      ir.CreateStore(exct_value_0, store_loc_0);
+
+      auto exct_value_1 = ir.CreateExtractValue(lpad, 1);
+      auto store_loc_1 = ir.CreateAlloca(exct_value_1->getType());
+      ir.CreateStore(exct_value_1, store_loc_1);
+
+      // if `ctx.ea_to_block.count(entry->lp_ea) == 0`, the landing pad basic block
+      // might not be recovered. Throw a warning in such case.
+      LOG_IF(ERROR, !ctx.ea_to_block.count(entry->lp_ea))
+          << "Missing block at " << std::hex << entry->lp_ea
+          << " for exception landing pad from " << entry->start_ea << std::dec;
+
+      auto lp_entry = GetOrCreateBlock(ctx, entry->lp_ea);
+      ir.CreateBr(lp_entry);
+      ctx.lp_to_block[entry->lp_ea] = landing_bb;
+
+    } else {
+      std::stringstream ss;
+      ss << "catch_landingpad_" << std::hex << entry->lp_ea;
+      auto landing_bb = llvm::BasicBlock::Create(*gContext, ss.str(), ctx.lifted_func);
+      llvm::IRBuilder<> ir(landing_bb);
+      auto num_clauses = entry->type_var.size();
+      auto exn_type = llvm::StructType::get(llvm::Type::getInt8PtrTy(*gContext),
+                                            llvm::Type::getInt32Ty(*gContext), nullptr);
+#if LLVM_VERSION_NUMBER > LLVM_VERSION(3, 6)
+      auto lpad = ir.CreateLandingPad(exn_type, static_cast<unsigned int>(num_clauses), "catch.lpad");
+#else
+      auto personality = gModule->getFunction(PERSONALITY_FUNCTION);
+      auto lpad = ir.CreateLandingPad(exn_type, personality, static_cast<unsigned int>(num_clauses), "catch.lpad");
+#endif
+      auto catch_all = false;
+
+      // The type indices in exception table are reversed for some of the binaries. In such case, the
```

Comment:
Add an issue to track this and update the comment to reference the issue number.